File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = float("inf")
        max_prof = 0
        #res = [] ## for debug only
        for price in prices:
            min_price = min(min_price,price)
            profit = price - min_price
            max_prof = max(max_prof, profit)
            #res.append(max_prof) # not required, for debug only
        return max_prof
        


## 7,1,5,3,6,4 - maxprofit = 1 par stock buy karun and 6 par sell kar du
# sabse min price lo, profit nikalo, usko max_profit kardo
## max_prof = 0, 0, 4, 4, 5, 

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int maxi = 0;
        int n = prices.size();
        int min_price = INT_MAX;
        for(int i=0; i<n; i++){
            min_price = min(min_price, prices[i]);
            profit = prices[i] - min_price;
            maxi = max(maxi, profit);
        }
        return maxi;
    }
};